aspnetcore/performance/caching/middleware.md

```diff
@@ -132,7 +132,7 @@ When testing and troubleshooting caching behavior, a browser may set request hea
 
 * The request must result in a server response with a 200 (OK) status code.
 * The request method must be GET or HEAD.
-* Terminal middleware must not process the response prior to the Response Caching Middleware.
+* [Terminal middleware](xref:fundamentals/middleware/index#built-in-middleware) must not process the response before Response Caching Middleware in the middleware request processing pipeline. For more information, see <xref:fundamentals/middleware/index>.
```

Comment:
I'm not sure why this comment is here. This isn't specific to response caching, this is how ordering works with any middleware.

Reply:
This isn't a general informational list. It's in the [Response Caching Topic (here)](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/middleware?view=aspnetcore-2.2#conditions-for-caching). This is a list of conditions that must be true for caching to occur. This line is just calling it out for this middleware's ordering. Actually, this was the line that confused the reader and generated the whole PR.

Reply:
Would they be less confused if we removed the line?

Reply:
**_I say nae_** ... if a user of this middleware places it after something else ... some middleware of their own perhaps that shoots a response, it won't work. That was the question. The reader is asking for clarification on the ordering of this middleware.

This is also part of a *larger docs concern* that I have based on the number of questions that we've had on middleware ordering. We closed [Confirm middleware order notes appear in individual topics \#8706](https://github.com/aspnet/Docs/issues/8706) due to a lack of staffing (or cloning ... since David refused to give me his cloning tech on the grounds that *one Rex is more than enough!* 😝).

This is a list of **conditions** for the middleware to work, and **_it's order is a condition of it working_**.

What's your objection to this line? It's clear now and links to the right info.

Reply:
Let's use the same language from the linked doc. "The middleware must be placed before components that require compression." It's simpler and positive rather than negative.

Reply:
ok

Reply:
> The middleware must be placed before components that require compression.

Let's say *where* and flesh out *before*. I'm scared of using "components," so let's say "middleware scenarios" (given that we can't use the word "features" any longer).

> In `Startup.Configure`, Response Caching Middleware must be placed before middleware scenarios that require compression. For more information, see \<xref:fundamentals/middleware/index>.

Reply:
Scenarios?
"before middleware that require compression"?

Reply:
Yeah ... we're limited given that "features" is a *banned word* now. lol

Stand-by ........................

Reply:
The subject-verb agreement *sounds* :ear: a hair funky given that "middleware" is plural here but sounds/assumed singular when read. No worries. I'm sure that readers will survive. :smile: